### PR TITLE
Add sorting functionality to listdatahandler.

### PR DESCRIPTION
**Pull request by MikolJiron**
Adds a new template that contains two dropdown which are used to indicate sort parameter and direction.
Sort colleges by default. When a change is made to sort parameter or sort direction, the list is rerendered and a query is made with the new parameter changes. The main thing that needs to be changed to support sorting for scholarships is scholarshiplistdatahandler (the backend also needs to be changed slightly: sorting by a param is already supported by the backend, but specifying sort direction is not).

### CODE REVIEW

File: src/main/webapp/js/finscholar/commonlistview/commonlistview.js [hunk 1]
```diff
@@ -134,12 +152,22 @@ class CommonListView extends BasicView {
...
+      asc: ASCENDING,
+      desc: DESCENDING}));
     super.resetAndUpdate();
     this.initJsaction_();
     // tableContainer.innerHTML = commonlistview({pagetype: this.optionTag_});
     this.container_ = googDom.getElement(ITEM_CONTAINER_ID);
     this.statusBar_ = googDom.getElement(STATUS_BAR_ID);
+    this.sortBySelector = googDom.getElement(SORT_BY_SELECTOR_ID);
+    this.sortBySelector.value = this.sortBy;
+    this.sortBySelector.addEventListener('change', this.bindedSelectorHandler_);
```
Comment (Lorraine-Lyu) [inline src/main/webapp/js/finscholar/commonlistview/commonlistview.js]: Wow I'm surprised that addEventListener works here. I not sure whether it's better than jsaction, so let's first keep this here and make changes if needed later.

**Review by Lorraine-Lyu — APPROVED** [review]

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Getting these linter messages:

```
src/main/webapp/js/datahandlers/collegelistdatahandler.js:22: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'ASCENDING' is never referenced in this file
  22| const {ACCEPTANCE_RATE, ACT_SCORE, ASCENDING, CollegeQueryBuilder, DESCENDING, ID, NAME} = 
                                         ^^^^^^^^^

src/main/webapp/js/datahandlers/collegelistdatahandler.js:22: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'DESCENDING' is never referenced in this file
  22| const {ACCEPTANCE_RATE, ACT_SCORE, ASCENDING, CollegeQueryBuilder, DESCENDING, ID, NAME} = 
                                                                         ^^^^^^^^^^

src/main/webapp/js/datahandlers/collegequerybuilder.js:89: WARNING - [JSC_MISSING_RETURN_JSDOC] Function with non-trivial return must have JSDoc indicating the return type.
  89|   static buildSortedCollectionEndpoint(batchIndex, itemsPerBatch, sortParam, sortDirection) {
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  90|     return COLLEGE_LIST_ENDPT.concat(COLLEGES, AND, PRIVATE, AND, QUERY_FIELDS, 
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  95|       AND, API_KEY_FIELD, COLLEGE_API_KEY);
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  96|   }
      ^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const JsactionActionFlow = goog.require('jsaction.ActionFlow');
const JsactionDispatcher = goog.require('jsaction.Dispatcher');
const JsactionEventContract = goog.require('jsaction.EventContract');
const googDom = goog.require('goog.dom');
const googSoy = goog.require('goog.soy');
const {ASCENDING, DESCENDING, SORT_PARAMS_MAP} = goog.require('datahandlers.collegequerybuilder');
const {BasicView} = goog.require('basicview');
const {ListDataHandler} = goog.require('datahandlers.listdatahandler');
const {commonlistview, endoflist, listitems, loading} = goog.require('finscholar.commonlistview.templates');

  19| const JsactionActionFlow = goog.require('jsaction.ActionFlow');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:86: WARNING - [JSC_MUST_HAVE_TRAILING_UNDERSCORE] Private property sortBy should end with '_'
  86|     this.sortBy = Array.from(SORT_PARAMS_MAP)[0][0];
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:89: WARNING - [JSC_MUST_HAVE_TRAILING_UNDERSCORE] Private property sortOrder should end with '_'
  89|     this.sortOrder = ASCENDING;
          ^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:92: WARNING - [JSC_MUST_HAVE_TRAILING_UNDERSCORE] Private property sortBySelector should end with '_'
  92|     this.sortBySelector = null;
          ^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:95: WARNING - [JSC_MUST_HAVE_TRAILING_UNDERSCORE] Private property sortOrderSelector should end with '_'
  95|     this.sortOrderSelector = null;
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
More linter messages:

```
src/main/webapp/js/finscholar/commonlistview/commonlistview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const JsactionActionFlow = goog.require('jsaction.ActionFlow');
const JsactionDispatcher = goog.require('jsaction.Dispatcher');
const JsactionEventContract = goog.require('jsaction.EventContract');
const googDom = goog.require('goog.dom');
const googSoy = goog.require('goog.soy');
const {ASCENDING, DESCENDING, SORT_PARAMS_MAP} = goog.require('datahandlers.collegequerybuilder');
const {BasicView} = goog.require('basicview');
const {ListDataHandler} = goog.require('datahandlers.listdatahandler');
const {commonlistview, endoflist, listitems, loading} = goog.require('finscholar.commonlistview.templates');

  19| const JsactionActionFlow = goog.require('jsaction.ActionFlow');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Review by CarlosM10011 — APPROVED** [review]

File: src/main/webapp/js/finscholar/commonlistview/commonlistview.soy [hunk 2]
```diff
@@ -51,4 +63,24 @@
 
 {template .endoflist}
   <p> This is the end of the list. </p>
+{/template}
+
+{template .sortByComponent}
+  {@param paramsToSortBy : map<string, string>}
+  <label for="sortBy">Choose Parameter to Sort By: </label>
```
Comment (tinagrasso) [inline src/main/webapp/js/finscholar/commonlistview/commonlistview.soy]: nit: can you make this more concise? Just "Sort by" would be fine.

File: src/main/webapp/js/finscholar/commonlistview/commonlistview.soy [hunk 3]
```diff
@@ -51,4 +63,24 @@
...
+    {for $key in mapKeys($paramsToSortBy)}
+      <option value={$key}>{$paramsToSortBy[$key]}</option>
+    {/for}
+  </select>
+{/template}
+
+{template .sortOrderComponent}
+  {@param ascending : string}
+  {@param descending : string}
+  <label for="sortOrder">Choose Sort Order: </label>
```
Comment (tinagrasso) [inline src/main/webapp/js/finscholar/commonlistview/commonlistview.soy]: nit: same as above. "Order" would work just as well and unclutter the UI. Optionally a checkbox would with "Asc" as the label would do the trick.

**Review by tinagrasso — CHANGES_REQUESTED** [review]
Can you provide a screenshot of what this looks like? I remember there being some whitespace issues between the two controls.

**Review by hzimz — APPROVED** [review]

**Review by tinagrasso — APPROVED** [review]

**Comment by MikolJiron** [comment]
Screen Shot 2020-08-03 at 1 31 26 PM
I fixed the spacing as well.

**Pull request merged**